public/app/plugins/panel/geomap/GeomapPanel.tsx

```diff
@@ -250,24 +249,26 @@ export class GeomapPanel extends Component<Props, State> {
     pointerMoveListener(evt, this);
   };
 
-  initMapView = (config: MapViewConfig): Array<View | undefined> => {
-    let view = new View({
+  initMapView = (config: MapViewConfig): View | undefined => {
+    const view = new View({
       center: [0, 0],
       zoom: 1,
       showFullExtent: true, // allows zooming so the full range is visible
     });
-
     // With shared views, all panels use the same view instance
     if (config.shared) {
       if (!sharedView) {
         sharedView = view;
+        this.initViewExtent(view, config);
+        return view;
       } else {
-        view = sharedView;
+        this.initViewExtent(sharedView, config);
+        return sharedView;
       }
+    } else {
+      this.initViewExtent(view, config);
+      return view;
     }
-    this.initViewExtent(view, config);
-
-    return [sharedView, view];
   };
```

Comment:
hm I think we can still keep the end of this function the same (pretty much) i.e.

```
this.initViewExtent(view, config);

return view;
```

and just define `view` as a let var and assign it accordingly? unless that doesn't work for some reason?